src/NuGet.Core/NuGet.Protocol/Repository.cs

```diff
@@ -27,7 +39,59 @@ public class RepositoryFactory
 
         public class ProviderFactory
         {
-            // Methods are added by extension
+            public virtual IEnumerable<Lazy<INuGetResourceProvider>> GetCoreV3()
```

Comment:
It's not necessary to use a different name to avoid confusion with the extension method.

From https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods#binding-extension-methods-at-compile-time:

> An extension method with the same name and signature as an interface or class method will never be called. At compile time, extension methods always have lower priority than instance methods defined in the type itself. In other words, if a type has a method named Process(int i), and you have an extension method with the same signature, the compiler will always bind to the instance method. When the compiler encounters a method invocation, it first looks for a match in the type's instance methods. If no match is found, it will search for any extension methods that are defined for the type, and bind to the first extension method that it finds.